### PR TITLE
added lang attribute to html file

### PR DESCRIPTION
**Pull request by NidhiKJha**
This pull request adds `lang` attribute in HTML files

### CODE REVIEW

**Comment by NidhiKJha** [comment]
@IanLawrence please review my pull request :smile:

**Comment by anubhavsingh16** [comment]
@NidhiKJha Is it GSoC PR?

**Comment by pnabutovsky** [comment]
Unfortunately it's not that simple. The current language is dynamic. On the server side (in the instance of mailer htmls) it may come from `I18n.locale`, but really it should be injected from the client side. The User API (which is already being called and cached in Redux state) comes back with a `locale` property. That locale may optionally be overridden by a manual choice of the locale selection the widget.

**Comment by jtoliver** [comment]
When users switch locales in primero, you will have to also change lang via react/js. Take a look at app/javascript/components/i18n/provider.js in the changeLocale method. You can do it there

**Comment by NidhiKJha** [comment]
Hi @pnabutovsky @jtoliver I looked into it, and noticed on my local development server, the redux state for locale is `null`. 
WhatsApp Image 2020-06-12 at 3 53 05 PM

also I noticed that we change the global window object when setting redux locale and hence I can access the current locale via `window.I18n.locale` on the deployed link
Screenshot_20200612_160158

I wanted to ask which approach should I take since I'm not sure why the redux locale is null. One more thing, I was able to change language in deployed app but in the development mode I am not getting that option
Screenshot from 2020-06-12 16-14-12

**Comment by jtoliver** [comment]
> Hi @pnabutovsky @jtoliver I looked into it, and noticed on my local development server, the redux state for locale is `null`.
> WhatsApp Image 2020-06-12 at 3 53 05 PM
> 
> also I noticed that we change the global window object when setting redux locale and hence I can access the current locale via `window.I18n.locale` on the deployed link
> Screenshot_20200612_160158
> 
> I wanted to ask which approach should I take since I'm not sure why the redux locale is null. One more thing, I was able to change language in deployed app but in the development mode I am not getting that option
> Screenshot from 2020-06-12 16-14-12

Use i18n.locale, and you might be missing a step before running rails s. Try running `rails primero:i18n_js` then start the server to see if that fixes it.

**Comment by NidhiKJha** [comment]
> Use i18n.locale, and you might be missing a step before running rails s. Try running `rails primero:i18n_js` then start the server to see if that fixes it.

After running `rails primero:i18n_js` I got this error
Screenshot from 2020-06-12 22-38-40 (1)

For i18n.locale, which approach should I use redux or window

**Comment by jtoliver** [comment]
Use window.I18n.locale

**Comment by NidhiKJha** [comment]
@jtoliver I have made changes. Can you let me know if my approach is in the right direction? Also, I have added test files but I am stuck with this error. 

Screenshot from 2020-06-16 18-08-28

Can you please let me know how can I correctly mount and access the `changeLocale` method in the test.

File: app/javascript/components/i18n/provider.unit.test.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      <Provider store={store()}>
+        <I18nProvider>
+          <div>wrapped children</div>
+        </I18nProvider>
+      </Provider>
+    );
+
+    console.log(component.find(I18nProvider).first().instance()); // null
+
+    component.find(I18nProvider).first().instance().changeLocale(newLocale);
```
Comment (jtoliver) [inline app/javascript/components/i18n/provider.unit.test.js:33]: Is component.find(I18nProvider).first() returning the correct component?

Comment (NidhiKJha) [reply app/javascript/components/i18n/provider.unit.test.js:33]: No, it's not, it's `null`, I tried different variations and they're all `null` and I'm not sure how I should be mounting it.

**Comment by NidhiKJha** [comment]
@pnabutovsky "en" is only the initial value given. I've made it dynamic by setting the lang with `document.documentElement.lang`

This all is done on the client side but I'm not sure how it could be made dynamic on the server side (inside the `.html.erb` files)

The lang only changes when a users toggles it, by default it's "en".

**Comment by pnabutovsky** [comment]
@NidhiKJha The changes I proposed take care of it on the server side. The majority of the rails templates you are changing are actually used for html email, not for explicit browser rendering, but this seems like a good idea across the board.

**Comment by NidhiKJha** [comment]
Thanks for the suggestions @pnabutovsky . I've made the suggested changes. Please review it  :)

**Pull request merged**